### PR TITLE
Update default branch to trunk

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR prepares the repository for the default branch rename from `develop` to `trunk`. It updates the release toolkit to `2.3.0` and adds the new environment variable for the default branch.

Please approve this PR, but **do not merge it**.

_P.S:_ There is a good chance that jumping from `1.3` of release toolkit to `2.3` will result in some issues. However, tooling in the Simplenote repositories are outdated and will likely need to be re-visited when the development continues regardless of this, so I am in favor of doing the update together with the default branch rename.

### CODE REVIEW

**Comment by oguzkocer** [comment]
```
Fetching git@github.com:Automattic/SimplenoteInterlinks-Swift.git

xcodebuild: error: Could not resolve package dependencies:
  The server SSH fingerprint failed to verify.
```

@Automattic/owl-team I unfortunately have no idea what this CI error is about. Can anyone help me address it?

File: Gemfile.lock [hunk 1]
```diff
@@ -153,7 +153,7 @@ GEM
       xcpretty-travis-formatter (>= 0.0.3)
     fastlane-plugin-appcenter (1.6.0)
     fastlane-plugin-sentry (1.6.0)
-    fastlane-plugin-wpmreleasetoolkit (1.3.1)
+    fastlane-plugin-wpmreleasetoolkit (2.3.0)
```
Comment (AliSoftware) [inline Gemfile.lock:156]: Just to note that I [double-checked our list of Breaking Changes when bumping from `1.x.x` to `2.x.x`](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md#breaking-changes-1) and confirmed that none of those breaking changes are impacting `simplenote-macos`, so it's ok not to have any other changes about this major version bump in this PR other than the `ENV['FL_RELEASE_TOOLKIT_DEFAULT_BRANCH'] = 'trunk'`  👍

**Review by AliSoftware — APPROVED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (oguzkocer) [reply Gemfile.lock:156]: Thank you for checking that @AliSoftware 🙇‍♂️

**Comment by AliSoftware** [comment]
> ```
> Fetching git@github.com:Automattic/SimplenoteInterlinks-Swift.git
> 
> xcodebuild: error: Could not resolve package dependencies:
>   The server SSH fingerprint failed to verify.
> ```
> 
> @Automattic/owl-team I unfortunately have no idea what this CI error is about. Can anyone help me address it?

This is strange, I was about to say that this is likely because Swift Package Manager (SPM) fails to clone the git repo of that dependency because of SSH known hosts not being pre-configured for it, but I see that [the `.circleci/config.yml` already has the patch we usually use to handle that](https://github.com/Automattic/simplenote-macos/blob/develop/.circleci/config.yml#L21-L25), so this feels odd. Maybe a previous entry in the pre-existing `~/.ssh/known_hosts` of the VM (with an old SSH fingerprint) takes precedence? Might be worth printing the `~/.ssh/known_hosts` during the CI steps to help debug, and maybe delete the `~/.ssh/known_hosts` before running that `for ip in $(dig …)` workaround?

**Comment by oguzkocer** [comment]
I am going to have to ignore the CI issue and merge this PR today as all default branch updates will happen together. We can address the issue separately.

**Pull request merged**